DEV-684 TextField and Slider component with light/dark mode

willmccormick24 opened the pull request:
I created the `theme-context` file to manage light/dark mode. Also, I created a new color file with the Figma redesign color styles.

yuushazen commented on src/components/insync/text-field/textField.tsx:
```diff
@@ -0,0 +1,172 @@
+		<TextFieldStyled isDark={isDark}>
+			<label className="label">{label}</label>
+			<div
+				className={`input-wrapper ${isHovered ? 'hovered' : ''} ${isActive ? 'active' : ''} ${error ? 'error' : ''} ${
```
Let's use the `classnames` library to make this simpler

Ericky14 replied:
Let's use the `classnames` library to make this simpler

yuushazen commented on src/theme-context.tsx:
```diff
@@ -0,0 +1,30 @@
+
+const ThemeContext = createContext<ThemeContextType | undefined>(undefined);
+
+export const ThemeProvider: React.FC<ThemeProviderProps> = ({ children }) => {
```
This is not how we should implement themes, we should use the styled components theming approach
https://styled-components.com/docs/advanced

Ericky14 replied:
This is not how we should implement themes, we should use the styled components theming approach
https://styled-components.com/docs/advanced

yuushazen commented on src/components/insync/text-field/textField.tsx:
```diff
@@ -0,0 +1,172 @@
+	--background-color: ${props => (props.isDark ? colors.deepBlack : colors.white)};
+	--flip-color: ${props => (props.isDark ? colors.white : colors.trueBlack)};
+	--8-2-color: ${props => (props.isDark ? colors.gray8 : colors.gray2)};
+	--2-8-color: ${props => (props.isDark ? colors.gray2 : colors.gray8)};
```
We should definately not be doing this, you should be having one property which changes values based on which theme we are using, such as `primaryColor` instead of `gray2` and `gray8`

Ericky14 replied:
We should definately not be doing this, you should be having one property which changes values based on which theme we are using, such as primaryColor instead of gray2 and gray8

yuushazen reviewed (changes requested):
Please refactor how themes are implemented

Ericky14 reviewed (changes requested):
Please address comments

willmccormick24 commented:
Light Mode:
<img width="1107" alt="Screenshot 2023-09-07 at 3 41 41 PM" src="https://github.com/rebuschain/rebus.app.web/assets/143646590/888b9872-5b0d-49be-a8e2-c9b07867609f">

Dark Mode:
<img width="1096" alt="Screenshot 2023-09-07 at 3 43 54 PM" src="https://github.com/rebuschain/rebus.app.web/assets/143646590/8f4a495b-ed47-492e-b98a-928ca0d0ffbc">

willmccormick24 commented:
> 

Ok I'll change the theme implementation to be in line with https://styled-components.com/docs/advanced and address your comments! Sorry I did the theming the way I've done in the past but I'll correct. I just posted screenshots of TextField and Slider redesign with the two different themes.

nicolam1 reviewed (changes requested):
The way the theme is handled needs to be improved.

willmccormick24 commented:
Theming now uses styled-components and I changed how properties are implemented and I addressed Ericky's other comments. No difference in the UI between commits so screenshots are still valid.

Ericky14 commented on src/components/common/slider.tsx:
```diff
@@ -0,0 +1,93 @@
+}
+
+const Slider: React.FC<SliderProps> = ({ toggleTheme, isDarkTheme }) => {
+	const [isToggled, setToggled] = useState(isDarkTheme);
```
This component should not have a local state, the slider value should depend on whether the theme is set to dark or not

Ericky14 commented on src/theme.tsx:
```diff
@@ -0,0 +1,47 @@
+	error: colors.error,
+};
+
+export const useTheme = () => {
```
I dont get how this is working, it has a local theme state when in reality it should be coming from the StyledComponents theme context

Ericky14 commented on src/pages/assets/index.tsx:
```diff
@@ -10,12 +10,38 @@ import { AssetBalancesList } from './asset-balances-list';
+
+Place inside return below to test Textfield, Slider, and:
+
+<ThemeProvider theme={theme === 'dark' ? darkTheme : lightTheme}>
```
Theme provider should be added as a wrapper to the whole app

Ericky14 reviewed (changes requested):
Please address comments

willmccormick24 commented:
Ok I think I addressed these comments, I'm trying to base the light/dark mode on our similar project. I should be able to commit my changes soon 👍

willmccormick24 commented:
Ok I made the theming global, stored changes in cookies to persist through sessions, got rid of the local state for the slider, and addressed the other comments 👍

nicolam1 reviewed (approved):
LGTM
